code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityExtension.java

```diff
@@ -103,26 +108,85 @@ protected String getVersion() {
 		return IdentityConstants.EXTENSION_VERSION;
 	}
 
-
 	/**
-	 * Call this method with the EventHub's Boot event to handle the boot operation of the {@code Identity} Extension.
-	 * <p>
-	 * On boot share the initial identities loaded from persistence to XDM shared state.
-	 *
-	 * @param event the boot {@link Event}
+	 * Adds an event to the event queue and starts processing the queue.
+	 * @param event the received event to be added in the events queue; should not be null
 	 */
-	void handleEventHubBoot(final Event event) {
+	void processAddEvent(final Event event) {
+		if (event == null) {
+			return;
+		}
 
-		// share the initial XDMSharedState on bootUp
-		final Map currentIdentities = state.getIdentityProperties().toXDMData(false);
+		cachedEvents.add(event);
+		processCachedEvents();
```

Comment:
This in-memory queue is required to queue incoming events while we wait for initial hub&identity shared state updates and to complete the bootup.